src/lammpsio/snapshot.py

```diff
@@ -107,18 +107,64 @@ def from_hoomd_gsd(cls, frame):
             if numpy.any(snap.molecule < 0):
                 warnings.warn("Some molecule IDs are negative, remapping needed.")
 
-        if (
-            frame.bonds.N > 0
-            or frame.angles.N > 0
-            or frame.dihedrals.N > 0
-            or frame.impropers.N > 0
-        ):
-            warnings.warn("Conversion of topology from gsd is not supported")
-        type_map = {typeid + 1: i for typeid, i in enumerate(frame.particles.types)}
-
-        return snap, type_map
-
-    def to_hoomd_gsd(self, type_map=None):
+        # set particle label
+        label_map_particle = None
+        if frame.particles.types is not None:
+            label_map_particle = {
+                typeid + 1: i for typeid, i in enumerate(frame.particles.types)
+            }
+        snap.label = LabelMap(map=label_map_particle)
```

Comment:
I think this should only be set if the types are present (which they are basically all the time)
```suggestion
            snap.label = LabelMap(map=label_map_particle)
```